### DISCUSSION TITLE
Custom javascript block

### DISCUSSION THREAD

**Discussion by WilliamWCYoung** [postmanlabs/postman-flows#106] [Ideas]
At the moment I'm blocked re-creating my flow because I need to do a data transformation similar to:

```
[
{"foo": "a", "bar": 1},
{"foo": "b", "bar": 2},
]
```
=>
```
{
"a": 1,
"b": 2,
"c": 3,
}
```

Where the first is data from one request and the second is data I want to send to the next. 

It feels like we need to have a block that can just run some javascript like the test section on requests. I don't want to use the test section as I may want to use the request in different ways at different parts of the flow.

Thank you! The tool is awesome

**Comment by saswatds** [comment]
Hey, @WilliamWCYoung thanks for your feedback. We are currently working on data manipulation as a feature of flows itself. With flows, we are trying to remove javascript as a dependency on the whole (maintaining a sandbox is hard if we embed another VM into the flow language). Everything you are capable of doing with javascript will be possible in flows as well, possibly even more easily.

Stay tuned.

**Reply by WilliamWCYoung** [reply]
Thank you - that was the intention I imagined, and will obviously make this tool really nice to use! I am unsure how the product will able to be as extensive as javascript but am excited to see.

Is there a roadmap of blocks that are being proposed, and can I make a suggestion of a block if it's missing for a use-case?
